File: JAVA/2779. Maximum Beauty of an Array After Applying Operation.java
class Solution {
    public int maximumBeauty(int[] n, int k) {
        Arrays.sort(n);
        int l=0,m=0;
        for(int i=0;i<n.length;i++){
            while(l<n.length&& n[i]-n[l]>2*k)l++;
            m=Math.max(m,i-l+1);
        }
        return m;
    }
}

File: C++/2779. Maximum Beauty of an Array After Applying Operation.cpp
class Solution {
public:
    int maximumBeauty(vector<int>& n, int k) {
       sort(n.begin(), n.end());
        int m = 0, l = 0;
        for (int r = 0; r < n.size(); r++) {
            while (n[r] - n[l] > 2 * k) l++;
            m = max(m, r - l + 1);
        }
        return m;
    }
};